mobile/library/common/jni/BUILD

```diff
@@ -133,7 +133,9 @@ config_setting(
 # Main dynamic library for the Envoy Mobile aar
 cc_binary(
     name = "libenvoy_jni.so",
-    linkopts = select({
+    linkopts = [
+        "-ldl",
```

Comment:
Let's add a comment here which explains why we need thisand mentions that it's not covered by CI so that we don't accidentally delete it again :). Maybe something like:

  Without this option the app crashes when attempted to being launched on arm64 emulator which is not covered by CI.

Reply:
Might as well figure out how to cover it on CI. It's a launch crash so I'm sure that coverage will help other things too

Reply:
Agreed! Do you happen to know how plausible it would be to get arm64 coverage in CI?

Reply:
I assume so but I don't recall if we tried

Reply:
It _should_ be relatively easy - I assume we would need to specify an arm emulator in https://github.com/envoyproxy/envoy/blob/4fb3eeb1a12b9005133639e96425fef16816c2a4/mobile/ci/mac_start_emulator.sh#L5.

Reply:
[rafa](https://github.com/envoyproxy/envoy/issues/24974)